crates/uv-extract/src/stream.rs

```diff
@@ -93,6 +93,7 @@ pub async fn untar<R: tokio::io::AsyncBufRead + Unpin>(
     let decompressed_bytes = async_compression::tokio::bufread::GzipDecoder::new(reader);
     let mut archive = tokio_tar::ArchiveBuilder::new(decompressed_bytes)
         .set_preserve_permissions(false)
```

Comment:
@charliermarsh Do you remember why we have that her but not in the sync impl? Docs say "This flag is disabled by default" in both impls.

Reply:
I think this is a mistake / typo. I probably meant for this to be mtime.

Reply:
We probably should be preserving permissions, right?

Reply:
After https://github.com/astral-sh/uv/pull/1743, i think we should preserve the execute permission, but everything else (read, write and owner, as far as i can tell) should always be set to os defaults. We can adapt the same logic as for wheel for tars (#1743).

Reply:
Yeah I was wondering if we were missing that for tars.